File: Progammers/연습문제/Level 1/체육복/Solution.java
import java.util.Arrays;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int[] before = new int[n+1];
        int[] after = new int[n+1];
        Arrays.fill(before, 1);
        for (int l : lost) {
            before[l] -= 1;
        }
        for (int r : reserve) {
            before[r] += 1;
        }
        
        for (int i=1; i < n+1; i++) {
            int cnt = before[i];
            if (cnt == 0) {
                continue;
            } else if (cnt == 1) {
                after[i] = 1;
            } else {
                after[i] = 1;
                if (i != 1 && after[i-1] == 0) {
                    after[i-1] = 1;
                } else if (i != n && after[i+1] == 0) {
                    after[i+1] = 1;
                }
            }
        }

        int answer = 0;
        for (int res : after) {
            if (res == 1){
                answer++;
            }
        }
        return answer;
    }
}

File: Progammers/연습문제/Level 1/체육복/Solution.py
def solution(n, lost, reserve):
    before, after = [False] + [1] * n, [False] + [0] * n
    for l in lost:
        before[l] -= 1
    for r in reserve:
        before[r] += 1
    
    for i, cnt in enumerate(before):
        if cnt == 0:
            continue
        elif cnt == 1:
            after[i] = 1
        else: # cnt == 2:
            after[i] = 1
            if i != 1 and after[i-1] == 0:
                after[i-1] = 1
            elif i != n and after[i+1] == 0:
                after[i+1] = 1
    
    return after.count(1)


if __name__ == "__main__":
    n = 5
    lost = [2, 4]
    reserve = [1, 3, 5]
    print(solution(n, lost, reserve))
